src/osd/PG.cc

```diff
@@ -236,18 +236,18 @@ void PGPool::update(OSDMapRef map)
       (pi->get_snap_epoch() == map->get_epoch())) {
     updated = true;
     if (pi->maybe_updated_removed_snaps(cached_removed_snaps)) {
-      pi->build_removed_snaps(newly_removed_snaps);
-      if (cached_removed_snaps.subset_of(newly_removed_snaps)) {
-          interval_set<snapid_t> removed_snaps = newly_removed_snaps;
-          newly_removed_snaps.subtract(cached_removed_snaps);
-          cached_removed_snaps.swap(removed_snaps);
+      if (cached_deleting_snaps.subset_of(pi->get_deleting_snaps())) {
```

Comment:
I suspect that the subset_of logic is not quite right here, since deleting_snaps is expected to shrink when snaps are purged.